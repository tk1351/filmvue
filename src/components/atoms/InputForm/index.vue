<script setup lang="ts">
import { computed, ref } from 'vue'

const props = withDefaults(
  defineProps<{
    initialValue?: string
    placeholder?: string
    width?: string
  }>(),
  {
    initialValue: '',
    placeholder: '',
    width: '100%',
  }
)

interface Emits {
  (eventName: 'update:modelValue', value: string): void
}
const emit = defineEmits<Emits>()

const inputValue = ref(props.initialValue)

const computedValue = computed({
  get: () => {
    return inputValue.value
  },
  set: (value: string) => {
    inputValue.value = value
    emit('update:modelValue', inputValue.value)
  },
})
</script>

<template>
  <input
    v-model="computedValue"
    type="text"
    class="input-form"
    :placeholder="placeholder"
  />
</template>

<style scoped lang="scss">
.input-form {
  width: v-bind(width);
  height: 40px;
  padding: 0 20px;
  color: $main-color;

  &::placeholder {
    color: $sub-color;
  }
}
</style>
