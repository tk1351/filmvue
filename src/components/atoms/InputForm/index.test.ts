import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import InputForm from './index.vue'

describe('Testing InputForm Component', () => {
  it('InputFormコンポーネントが正しく表示されること', () => {
    const wrapper = mount(InputForm)
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('Props', () => {
    describe('initialValue', () => {
      it('inputタグのvalue属性に、正しく値が渡っていること', () => {
        const expectedValue = 'expectedValue'
        const wrapper = mount(InputForm, {
          props: {
            initialValue: expectedValue,
          },
        })
        expect(wrapper.find('input').element.value).toBe(expectedValue)
      })
    })

    describe('placeholder', () => {
      it('inputタグのplaceholder属性に、正しく値が渡っていること', () => {
        const expectedPlaceholder = 'expectedPlaceholder'
        const wrapper = mount(InputForm, {
          props: {
            placeholder: expectedPlaceholder,
          },
        })
        expect(wrapper.find('input').attributes('placeholder')).toBe(
          expectedPlaceholder
        )
      })
    })

    describe('Event', () => {
      it('inputへ値を入力した際に、update:modelValueイベントが実行されること', async () => {
        const wrapper = mount(InputForm)
        await wrapper.find('input').setValue('test')
        expect(wrapper.emitted('update:modelValue')).toHaveLength(1)
      })
    })
  })
})
