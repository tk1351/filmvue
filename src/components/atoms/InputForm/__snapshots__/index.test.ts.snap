// Vitest Snapshot v1

exports[`Testing InputForm Component > InputFormコンポーネントが正しく表示されること 1`] = `
<input
  class="input-form"
  data-v-4617d264=""
  placeholder="search..."
  type="text"
/>
`;
